internal/tools/crosslink/crosslink.go

```diff
@@ -25,130 +25,64 @@
 package main
 
 import (
-	"encoding/json"
-	"errors"
-	"fmt"
-	"io"
 	"log"
 	"os"
-	"os/exec"
 	"path/filepath"
 	"strings"
-	"text/tabwriter"
 
 	"go.opentelemetry.io/otel/internal/tools"
+	"golang.org/x/mod/modfile"
 )
 
-type repo string
-
-type mod struct {
-	filePath   string
-	importPath string
-}
-
-func (r repo) findModules() (mods, error) {
-	var results []mod
-	err := filepath.Walk(string(r), func(path string, info os.FileInfo, err error) error {
-		if !info.IsDir() {
-			return nil
-		}
-
-		_, err = os.Stat(filepath.Join(path, "go.mod"))
-		if errors.Is(err, os.ErrNotExist) {
-			return nil
-		}
-		if err != nil {
-			return err
-		}
-
-		cmd := exec.Command("go", "mod", "edit", "-json")
-		cmd.Dir = path
-		out, err := cmd.Output()
-		if err != nil {
-			return err
-		}
-
-		var result struct {
-			Module struct {
-				Path string
-			}
-		}
-		err = json.Unmarshal(out, &result)
-		if err != nil {
-			return err
-		}
-
-		results = append(results, mod{
-			filePath:   path,
-			importPath: result.Module.Path,
-		})
-		return nil
-	})
-
-	return results, err
-}
-
-type mods []mod
-
-func (m mods) print(w io.Writer) error {
-	tw := tabwriter.NewWriter(w, 0, 0, 1, ' ', 0)
-	if _, err := fmt.Fprintln(tw, "FILE PATH\tIMPORT PATH"); err != nil {
-		return err
-	}
-	for _, m := range m {
-		if _, err := fmt.Fprintf(tw, "%s\t%s\n", m.filePath, m.importPath); err != nil {
-			return err
-		}
-	}
-	return tw.Flush()
-}
-
-func (m mods) crossLink() error {
+func crossLink(m []*modfile.File) error {
 	for _, from := range m {
-		args := []string{"mod", "edit"}
-
+		basepath := filepath.Dir(from.Syntax.Name)
 		for _, to := range m {
-			localPath, err := filepath.Rel(from.filePath, to.filePath)
+			newPath, err := filepath.Rel(basepath, filepath.Dir(to.Syntax.Name))
 			if err != nil {
 				return err
```

Comment:
do we not miss `from.Cleanup()` call b/c of this return?

Reply:
Sure, but why does that matter? In that case we will never write `from` to disk because of this error, why run `from.Cleanup()`?